### PR TITLE
(Towards #379) ensure symbol table cleanly ignores operators

### PR DESCRIPTION
**Pull request by arporter**
This is a towards since it just fixes the crash in the symbol table. It does not attempt to extend the symbol table to support operators.

### CODE REVIEW

**Comment by arporter** [comment]
This branch parses all LFRic source without problems. Ready for a review from anyone.

File: src/fparser/two/tests/fortran2003/test_usestmt_r1109.py [hunk 1]
```diff
@@ -281,12 +369,12 @@ def test_use_internal_error2(f2003_create):
         ast.items = (None, None, content, None, None)
         with pytest.raises(InternalError) as excinfo:
             str(ast)
-        assert ("entry 2 should be a module name but it is " "empty") in str(
+        assert "entry 2 should be a module name but it is " "empty" in str(
```
Comment (sergisiso) [inline src/fparser/two/tests/fortran2003/test_usestmt_r1109.py:372]: Why is there a `" "` between 'is' an 'empty'? as black rearraged this incorrecty?

It happens in a few places:
src/fparser/two/tests/fortran2003/test_usestmt_r1109.py
372:        assert "entry 2 should be a module name but it is " "empty" in str(

src/fparser/two/tests/fortran2003/test_cray_pointer_decl.py
118:        assert ("'Items' entry 0 should be a pointer name but it is " "empty") in str(

src/fparser/two/tests/fortran2003/test_position_edit_desc_r1013.py
140:        "items[1] in Class Position_Edit_Desc method tostr() is " "empty or None"

src/fparser/two/tests/fortran2003/test_hollerith_item.py
120:        assert ("entry 0 should be a valid Hollerith string but it is " "empty") in str(

File: src/fparser/two/tests/fortran2003/test_rename_r1111.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    assert isinstance(obj, tcls), repr(obj)
+    assert str(obj) == "a => b"
+
+    obj = tcls("operator(.foo.)=>operator(.bar.)")
+    assert isinstance(obj, tcls), repr(obj)
+    assert str(obj) == "OPERATOR(.FOO.) => OPERATOR(.BAR.)"
+
+    obj = tcls("operator_1=>operator_2")
+    assert isinstance(obj, tcls), repr(obj)
+    assert str(obj) == "operator_1 => operator_2"
```
Comment (sergisiso) [inline src/fparser/two/tests/fortran2003/test_rename_r1111.py:58]: Can you assert for each of this cases that the child[0] has been given or not the "OPERATOR" descriptor.

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter Just a couple of small suggestions, the rest looks good.

File: src/fparser/two/tests/fortran2003/test_usestmt_r1109.py (hunk 1 above)
Comment (arporter) [reply src/fparser/two/tests/fortran2003/test_usestmt_r1109.py:372]: Weird. I think that must have been Black. I've fixed them now.

**Comment by arporter** [comment]
Brought to master and made suggested changes. Ready for another look now (CI permitting).

**Review by sergisiso — APPROVED** [review]
@arporter All comments addressed. PR ready to merge

**Pull request merged**